### PR TITLE
[CAS-231] Refactor BaseMessageListItemViewHolder

### PR DESCRIPTION
**Pull request by zsmb13**
In this PR:

- Converting the base ViewHolder and all but one of the descendants to Kotlin.
- Simplifying the `bind` methods and the adapter's constructor. Values previously passed in as parameters of `bind` to the ViewHolder implementations are now available from the ViewHolder factory, and are passed in as constructor parameters.

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    resId: Int,
+    viewGroup: ViewGroup,
+    private val style: MessageListViewStyle
+) : BaseMessageListItemViewHolder<DateSeparatorItem>(resId, viewGroup) {
+
+    private val tv_date: TextView = itemView.findViewById(R.id.tv_date)
+    private val iv_line_right: ImageView = itemView.findViewById(R.id.iv_line_right)
+    private val iv_line_left: ImageView = itemView.findViewById(R.id.iv_line_left)
+
+    private var messageListItem: DateSeparatorItem? = null
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt:28]: It seems that there is not any point to store such property in class

**Review by ogkuzmin — DISMISSED** [review]

File: library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt (hunk 1 above)
Comment (zsmb13) [reply library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt]: Nice catch, I totally missed that it was unused.

File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentViewHolderFactory.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+open class AttachmentViewHolderFactory {
+    lateinit var listenerContainer: ListenerContainer
+
+    open fun getAttachmentViewType(
+        attachmentItem: AttachmentListItem
+    ): Int {
+        return when (attachmentItem.attachment.type) {
+            null ->
+                MessageViewHolderFactory.GENERIC_ATTACHMENT
```
[the remark this answers is not in the document]
Comment (zsmb13) [reply library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentViewHolderFactory.kt]: I was so gonna do that, but somehow I forgot after all. Nice catch, thank you!

**Comment by tschellenbach** [comment]
Happy to see that java code disappear :)

**Review by ogkuzmin — APPROVED** [review]

File: library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        messageListItem: DateSeparatorItem,
+        position: Int
+    ) {
+        configDate(messageListItem)
+        applyStyle()
+    }
+
+    private fun configDate(messageListItem: DateSeparatorItem) {
+        val messageDate = messageListItem.date.time
+        val now = Date()
```
[the remark this answers is not in the document]
Comment (zsmb13) [reply library/src/main/java/com/getstream/sdk/chat/adapter/DateSeparatorViewHolder.kt:37]: Do you mean for testing purposes?

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**